core-rust/core-api-server/src/core_api/conversions/addressing.rs

```diff
@@ -227,13 +226,14 @@ fn to_mapped_substate_id(substate_id: SubstateId) -> Result<MappedSubstateId, Ma
 
     // Start body of method
     let entity_id_bytes = re_node_id_to_entity_id_bytes(&substate_id.0)?;
-    let substate_key_bytes = substate_offset_to_substate_key_bytes(&substate_id.1)?;
+    // TODO(code review): do we need to include the node module id in the MappedSubstateId too?
```

Comment:
My preference here is yes so that there is little abstraction between the Core API and the underlying ledger state but I think it mostly depends on what is the Gateway consumer ready to handle.

Reply:
alright, I consulted @KrzysztofLabus-RDX and we came up with `module_type` name for the new field; please see if this mapping is good enough.